warehouse/warehouse.go

```diff
@@ -1084,11 +1086,9 @@ func (wh *HandleT) getUploadsToProcess(ctx context.Context, availableWorkers int
 		upload.UseRudderStorage = warehouse.GetBoolDestinationConfig("useRudderStorage")
 
 		if !ok {
-			uploadJob := UploadJobT{
-				upload:   &upload,
-				dbHandle: wh.dbHandle,
-				stats:    wh.stats,
-			}
+			uploadJob := wh.uploadJobFactory.NewUploadJob(&model.UploadJob{
```

Comment:
That is a good point; my corrections were hacky to make the tests pass and not proper. I am addressing them now.

Regarding this, I would prefer not to create a dummy warehouse, but rather not a create a warehouse at all if the type is empty. It is still not ideal, but at least it would be more easier to discover it this way.

Reply:
I followed a different approach were the whManager is passed as param.